File: 1. Two Sum.cpp
#include<iostream>
#include<vector>
#include<utility>
using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> map;

    for (int i = 0; i < nums.size(); i++) {
      if (map.find(nums.at(i)) != map.end()) {
        return {map.at(nums.at(i)), i};
      }
      map[target - nums.at(i)] = i;
    }
    return {-1, -1};
  }
};

File: C#/1. Two Sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> dict = new Dictionary<int, int>();
        int size = nums.Length;
        for (int i = 0; i < size; i++)
        {
            if (dict.ContainsKey(nums[i])) 
            {
                return new int[] {i, dict[nums[i]]};
            }
            else
            {
                dict.TryAdd(target - nums[i], i);
            }
        }
        return new int[] {-1, -1};
    }
}